jekyll/_posts/2021-09-25-version-160-released.md

```diff
@@ -0,0 +1,817 @@
+Note that the `csources` repo used was changed to `csources_v1`, the new setup is designed to be forward and backward compatible.
+
+## Building from a CI setup
+We now have an API to be used in CI which abstracts the implementation details: `. ci/funs.sh && nimBuildCsourcesIfNeeded`; in fact all the existing CI pipelines have been refactored to use this, see
+[#17815](https://github.com/nim-lang/Nim/pull/17815).
+
+
+# Contributors to Nim 1.6
+
+Many thanks to our recurring and new [contributors](https://github.com/nim-lang/Nim/graphs/contributors?from=2020-10-16&to=2021-09-25&type=c),
+Nim is a community driven collaborative effort that welcomes all contributions, big or small.
+
+
+# Backward compatibility and preview flags
+
+Starting with this release, we've introduced preview flags of the form `-d:nimPreviewX`
+(e.g. `-d:nimPreviewFloatRoundtrip`); they allow users to opt-in a new stdlib/compiler behavior
+that will likely become default in the next or a future release.
+These staging flags are aimed at minimizing backward compatibility issues.
+We also introduced opt-out flags of the form `-d:nimLegacyX`,  e.g. `-d:nimLegacyCopyFile`,
+for cases where the default was changed to the new behavior,
+and an explicit opt-out is needed for some transition period.
+
+# Major new features
+With so many new features, pinpointing the most salient ones is a subjective exercise,
+but here are a select few:
+
+
+## Strict effects
+The effect system was refined and there is a new `.effectsOf` annotation that does
+  explicitly what was previously done implicitly. See the manual for details.
+  To write code that is portable with older Nim versions, use this idiom:
+
+```nim
+when defined(nimHasEffectsOf):
+  {.experimental: "strictEffects".}
+else:
+  {.pragma: effectsOf.}
+
+proc mysort(s: seq; cmp: proc(a, b: T): int) {.effectsOf: cmp.}
+```
+
+To enable the new effect system, use --experimental:strictEffects. See [#18777](https://github.com/nim-lang/Nim/pull/18777) and RFC [#408](https://github.com/nim-lang/RFCs/issues/408).
+
+
+## `iterable[T]`
+Added `iterable[T]` type class to match called iterators, which enables writing:
+```nim
+template fn(a: iterable) # or template fn[T](a: iterable[T])
+# instead of:
+template fn(a: untyped)
+```
+This solves a number of long standing issues related to iterators. In particular,
+iterable arguments can now be used with MCS, e.g. `iota(3).toSeq` now works.
+See PR [#17196](https://github.com/nim-lang/Nim/pull/17196) for additional details.
+
+
+## Private imports and private field access
+- A new import syntax `import foo {.all.}` now allows to import all symbols (public or private)
+  from `foo`. This can be useful for testing purposes or for more flexibility in project organization.
+- Added a new module `std/importutils`, and an API `privateAccess`, which allows access to private fields
+  for an object type in the current scope.
+Example:
+```nim
+from system as system2 {.all.} import ThisIsSystem
```

Comment:
Is this valid syntax? the `import ThisIsSystem` looks wrong

Reply:
actually there was a typo in what i wrote, I've now corrected it and also added an example for `privateImport` (note that the problem wasn't `import ThisIsSystem` but the placement of `{.all.}`; `from system {.all.} as system2 import nil` is valid and so is `from system {.all.} as system2 import ThisIsSystem `